### PR TITLE
Drop Mambaforge Python distribution, add warning regarding Anaconda

### PR DESCRIPTION
**Pull request by hoechenberger**
Mambaforge will be discontinued, so don't recommend it in our install docs anymore:
https://conda-forge.org/news/2024/07/29/sunsetting-mambaforge/

Also added a warning that Anaconda's lawyers might come after your if you use their packages in a commercial environment without paying a license fee.

### CODE REVIEW

**Comment by hoechenberger** [comment]
Rendered docs:
https://output.circle-artifacts.com/output/job/4e3f160b-7cc6-4e1f-8a23-7bec690d28d1/artifacts/0/html/install/manual_install_python.html

**Review by sappelhoff — APPROVED** [review]

File: doc/install/manual_install_python.rst [hunk 1]
```diff
@@ -8,10 +8,18 @@ Installing Python
...
+our pre-built :ref:`installers`, which use ``conda`` environments under the hood.
+
+.. warning::
+   Anaconda, the company that develops the Anaconda and Miniconda Python
+   distributions, is now strictly enforcing its end-user license agreement.
+   This means that if you're found using the software in a commercial setting,
+   you may be required to purchase a license. If you're unsure about your
+   usage, we highly recommend using Miniforge or our installers instead. These
+   options, by default, avoid Anaconda packages and rely on community-created
+   packages from the `conda-forge`_ project.
```
Comment (drammock) [inline doc/install/manual_install_python.rst:22]:
```suggestion
   Anaconda Inc., the company that develops the Anaconda and Miniconda Python
   distributions,
   `changed their terms of service <https://www.anaconda.com/blog/update-on-anacondas-terms-of-service-for-academia-and-research>`__
   in March of 2024. If you're unsure about whether your usage situation requires a paid
   license, we recommend using Miniforge or our pre-built installer instead. These
   options, by default, install packages only from the community-maintained `conda-forge`_
   distribution channel, and avoid the distribution channels covered by Anaconda's terms
   of service.
```

**Review by drammock — APPROVED** [review]
Thanks for keeping tabs on this situation @hoechenberger! suggested re-wording of the warning below.

**Review by mscheltienne — APPROVED** [review]

**Comment by drammock** [comment]
looks like the CI errors here should be resolved by #12928 
(too bad we don't have merge queues enabled)

**Pull request merged**